org.eclipse.xtext.xbase.tests/src/org/eclipse/xtext/xbase/tests/resources/ResourceStorageTest.java

```diff
@@ -163,4 +170,47 @@ public void testWriteAndLoad() throws Exception {
 		Assert.assertEquals(originalAdapter.sourceToTargetMap.size(), restoredAdapter.sourceToTargetMap.size());
 		Assert.assertEquals(originalAdapter.targetToSourceMap.size(), restoredAdapter.targetToSourceMap.size());
 	}
+	
+	@Test
+	public void testWriteAndLoadWithMissing() throws Exception {
+		String code = "{}";
+		XExpression file = this.expression(code);
+
+		assertEquals(2, file.eResource().getContents().size()); // Jvm model loaded
+
+		JvmGenericType type = ((JvmGenericType) file.eResource().getContents().get(1));
+		// add a synthetic constructor 
+		JvmConstructor jvmConstructor = TypesPackage.eINSTANCE.getTypesFactory().createJvmConstructor();
+		jvmConstructor.setSimpleName("Test");
+
+		type.getMembers().add(jvmConstructor);
+		associator.associate(file.eResource().getContents().get(0), jvmConstructor);
+
+		StorageAwareResource originalResource = ((StorageAwareResource) file.eResource());
+		ByteArrayOutputStream bout = new ByteArrayOutputStream();
+		resourceStorageFacade.setStoreNodeModel(false);
+		resourceStorageFacade.createResourceStorageWritable(bout).writeResource(originalResource);
+		
+		// Remove synthetic constructor to cause an addition of a "none" fragment
+		type.getMembers().remove(jvmConstructor);
```

Comment:
@dhuebner I can understand that it'S possible to break the serialization if the model is corrupted. I'd be curious though how that state was "achieved" in the real case.

Reply:
@szarnekow It happens often when switching a branch, sometimes it also happens during regular work/typing. What I see is that the broken Xtend file has an active annotation or an inner Xtend class has one. `@Data` or `@FinalFieldConstructor` for example manipulating constuctors of the JvmType, so I suppose it's a kind of race condition. The problem is that I can't see why the constructor eResource not in place when a `none` fragment was added.